config/v1/metric_descriptor.proto

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package config.v1;
+
+import "label_descriptor.proto";
+
+// Defines a metric type and its schema.
+//
+// A given metric is described by a set of labels. These labels represent the
+// different dimensions to associate with the metric. A given metric holds a
+// unique value for potentially any combination of these labels.
+//
+// Metric values can be integers, floating point numbers, strings, or booleans.
+//
+// Here's an example metric descriptor defined in a YAML configuration
+// file:
+//
+//    metrics:
```

Comment:
Mandar, what is "request.size" in your world? A fact, or a label produced by mapping facts, or something new?

In the API we have now, what I think you are trying to capture via elementary_measurement is handled imperatively by the caller. The MetricDescriptor type is really intended to describe the characteristic of a single metric value, not how these values are acquired. So the MetricDescriptor says "I want to hold integer values" and "I want to have different physical counters for each label I define". When someone calls Report, they provide the metric descriptor name, and a metric value (an integer). The mixer then derives the needed labels to properly 'route' the value, and forwards the result to the adapters.

I don't want to bind the definition of the metric with how the metric is acquired, those are separate concerns.